partner_phone_search/models/res_partner.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2018 - TODAY Serpent Consulting Services Pvt. Ltd.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    @api.model
+    def name_search(self, name, args=None, operator='ilike', limit=100):
+        if not args:
+            args = []
+        if name:
```

Comment:
If you first test for `if not name` and then just return super, the biggest part of your code can have one indentation level less.